src/gt4py/backend/gtc_backend/dace/backend.py

```diff
@@ -160,27 +160,24 @@ def generate_dace_args(self, gtir, sdfg):
                 symbols[f"__{name}_J_stride"] = str(array.shape[2])
                 symbols[f"__{name}_I_stride"] = str(array.shape[1] * array.shape[2])
             else:
-                data_ndim = len(array.shape) - sum(array_dimensions(array))
+                dims = [dim for dim, select in zip("IJK", array_dimensions(array)) if select]
+                data_ndim = len(array.shape) - len(dims)
 
                 # api field strides
                 fmt = "gt::sid::get_stride<{dim}>(gt::sid::sid_get_strides(__{name}_sid))"
+
```

Comment:
in the line above there is a bug: it should be `gt::sid::get_strides`. `sid_` is wrong.